### ISSUE TITLE
Onboarding experience for settings sync could be improved for screen reader users

### ISSUE THREAD

**Issue by Neurrone** [microsoft/vscode#120012]
/CC @isidorn 
Issue Type: <b>Bug</b>

TLDR: this technically works but the experience could be much improved and less confusing, it took me a few mins to figure out what was happening

1. Enable settings sync. After installing vs code insiders for the first time, the onboarding asks if I'd like to enable settings sync. I did so, and I'm assuming there is a way to do so on demand as well.
    
    edit: this step does work well with a screen reader and there aren't any problems here.
2. The first prompt asks if you want to sign in. However, the question is not read out by NVDA. While tabbing around, I found a text input button and a sign in button, and had to resort to advanced NVDA commands to figure out was going on.
3. I initially thought the text box was asking for some sort of username until I realized that its actually meant to filter the list of 5 categories of settings you can sync. That was very confusing because at least with a screen reader, its not obvious at all. Its also right next to the sign in button in tab order, which reinforced the impression that you're supposed to enter some sort of username
4. The selection state of the list of categories is not read as you move about them and press space to select / deselected with NVDA. Only the number of selected items is announced which is fine for this example, but would be problematic if the widget is used for more complex choices elsewhere in Vs Code
5. After hitting sign in, the next screen's question is also not read out at all. I noticed that the questions for subsequent screens is read out properly

VS Code version: Code - Insiders 1.55.0-insider (d06d2f1d6245ce00b1c36a9cd81a9087d225173e, 2021-03-26T14:01:14.867Z)
OS version: Windows_NT x64 10.0.19042

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-8550U CPU @ 1.80GHz (8 x 1992)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>protected_video_decode: enabled<br>rasterization: enabled<br>skia_renderer: enabled_on<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|15.86GB (9.41GB free)|
|Process Argv|--crash-reporter-id 2e199e84-faa5-42dd-9839-04326cd6a273|
|Screen Reader|yes|
|VM|0%|
</details>Extensions: none<details>
<summary>A/B Experiments</summary>

```
vsliv695:30137379
vsins829:30139715
vsliv368cf:30146710
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
vstry914:30244316
pythonvsdeb440:30224570
pythonvsded773:30223139
pythonvspyt875:30259475
pythonvsnew554:30265444
pythontb:30258533
pythonvspyt943:30277000
openwsldoc:30278403
vspre833cf:30267465
pythonptprofiler:30281269
vscnewfiletext:30280636
vshan820:30276952
vscoregs2cf2:30281057
vstes263:30280004

```

</details>

**Comment by isidorn** [comment]
@Neurrone thanks a lot for the great feedback. Some of this are general UI issues and some are related to settings sync.
@sandy081 would you like me to go through this flaw and try to capture it in smaller issues? Or are you already aware of some of the problems @Neurrone is pointing out here? Thanks

**Comment by sandy081** [comment]
@isidorn It would be great if you can evaluate the flow and give me feedback by issues with actions to be taken. Thanks a lot in advance.

**Comment by isidorn** [comment]
I just tied out this flow on Windows with NVDA

1. When I F1 > turn on settings sync, I get the following dialog "Synchronizing your settings is a preview feature, please read the documentation before turning it on" with three buttons amongs which is "Turn On". NVDA reads all these buttons nicely to me
2. This is an issue, I filled https://github.com/microsoft/vscode/issues/120176 to cover this 
3. Coverd with https://github.com/microsoft/vscode/issues/120176
4. Cover with https://github.com/microsoft/vscode/issues/120176
5. Same problem as 120176 basicaly the same quick pick dialog is used which is not accessible

@Neurrone this is great feedback. Can you just clarify 1) since that problem I could not reproduce
Everything else should be covered by the issue I created. So I think @sandy081 can look into it. @sandy081 let me know if you need more help.

**Comment by Neurrone** [comment]
@isidorn I'm sorry for the confusion, 1 is actually not a problem, that was part of the steps I followed and it does work well for me. I'll edit to make it clearer.

**Comment by isidorn** [comment]
@Neurrone great, so I suggest that we continue the discussion in the issue #120176 where I provided recommendation on how to fix this.
So I think this one can be closed. Hope that is ok.

**Comment by sandy081** [comment]
Thanks @isidorn
